lasagne/utils.py

```diff
@@ -337,105 +355,91 @@ def create_param(spec, shape, name=None):
         # `name` attribute of the variable, but the user may have already
         # named the variable and we don't want to override this.
         if spec.ndim != len(shape):
-            raise RuntimeError("parameter variable has %d dimensions, "
-                               "should be %d" % (spec.ndim, len(shape)))
+            raise ValueError("%s has %d dimensions, should be %d" %
+                             (err_prefix % "Theano variable", spec.ndim,
+                              len(shape)))
+        if not spec.name:
+            spec.name = name
         return spec
 
-    elif isinstance(spec, np.ndarray):
-        if spec.shape != shape:
-            raise RuntimeError("parameter array has shape %s, should be "
-                               "%s" % (spec.shape, shape))
-        return theano.shared(spec, name=name)
-
-    elif hasattr(spec, '__call__'):
-        arr = spec(shape)
-        try:
-            arr = floatX(arr)
-        except Exception:
-            raise RuntimeError("cannot initialize parameters: the "
-                               "provided callable did not return an "
-                               "array-like value")
-        if arr.shape != shape:
-            raise RuntimeError("cannot initialize parameters: the "
-                               "provided callable did not return a value "
-                               "with the correct shape")
-        return theano.shared(arr, name=name)
-
     else:
-        raise RuntimeError("cannot initialize parameters: 'spec' is not "
-                           "a numpy array, a Theano expression, or a "
-                           "callable")
+        if hasattr(spec, '__call__'):
+            raise TypeError("%s is not a numpy array or a Theano expression" %
+                            (err_prefix % "value"))
+        else:
+            raise TypeError("%s is not a numpy array, a Theano expression, "
+                            "or a callable" % (err_prefix % "spec"))
 
 
 def unroll_scan(fn, sequences, outputs_info, non_sequences, n_steps,
                 go_backwards=False):
-        """
```

Comment:
Interesting that nobody noticed this before. But anyway, this shouldn't be part of your PR! Please leave the indentation for `unroll_scan` as it was.